spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAnnotationDrivenConfiguration.java

```diff
@@ -65,12 +68,19 @@ public SimpleRabbitListenerContainerFactoryConfigurer rabbitListenerContainerFac
 
 	@Bean
 	@ConditionalOnMissingBean(name = "rabbitListenerContainerFactory")
-	public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(
-			SimpleRabbitListenerContainerFactoryConfigurer configurer,
+	public AbstractRabbitListenerContainerFactory<?> rabbitListenerContainerFactory(
```

Comment:
II think I'd prefer two definitions with a `@ConditionalOnProperty` that translates that if/else below. We prefer to expose the concrete type.